erpnext/selling/doctype/sales_order_item/sales_order_item.json

```diff
@@ -542,7 +542,7 @@
    "collapsible": 0, 
    "depends_on": "eval:doc.margin_type && doc.price_list_rate", 
    "fieldname": "total_margin", 
-   "fieldtype": "Float", 
+   "fieldtype": "Currency", 
```

Comment:
@rohitwaghchaure You have to change fieldtype for margin in all other doctypes as well.